src/view/gitContentProvider.ts

```diff
@@ -29,9 +29,13 @@ export class GitContentProvider implements vscode.TextDocumentContentProvider {
 		}
 
 		const absolutePath = pathLib.join(this.repository.rootUri.fsPath, path).replace(/\\/g, '/');
-		let content = await this.repository.show(commit, absolutePath);
-
-		if (!content) {
+		let content: string;
+		try {
+			content = await this.repository.show(commit, absolutePath);
+			if (!content) {
```

Comment:
`content` can be `''`, which makes `!content` true.

Reply:
yeah I'm not sure if we handle the case where the file content is actually '' correctly. this should be the same behavior as before, where we use the fallback to see if the file has been deleted or added